lib/dockerregistry/inventory.go

```diff
@@ -1229,6 +1431,53 @@ func MkReadRepositoryCmdReal(
 	return &sh
 }
 
+// MkReadManifestListCmdReal creates a stream.Producer which makes a real call
+// over the network to read ManifestList information.
+func MkReadManifestListCmdReal(
+	sc *SyncContext,
+	gmlc GManifestListContext) stream.Producer {
+
+	var sh stream.HTTP
+
+	tokenKey, domain, repoPath := GetTokenKeyDomainRepoPath(
+		gmlc.RegistryContext.Name)
+
+	endpoint := fmt.Sprintf(
+		"https://%s/v2/%s/%s/manifests/%s",
+		domain,
+		repoPath,
+		gmlc.ImageName,
+		gmlc.Tag)
+
+	httpReq, err := http.NewRequest("GET", endpoint, nil)
+
+	// Without this, GCR responds as we had used the "Accept:
+	// application/vnd.docker.distribution.manifest.v1+prettyjws" header.
+	httpReq.Header.Add("Accept", "*/*")
+
+	if err != nil {
+		klog.Fatalf(
```

Comment:
Idea for the future: Maybe tackling stream.Producer is a reasonable place to start in terms of making the code simpler - can we just return a `([]byte, error)`

Reply:
Ack.